src/clarity-angular/utils/focus-trap/focus-trap.directive.ts

```diff
@@ -3,17 +3,32 @@
  * This software is released under MIT license.
  * The full license information can be found in LICENSE in the root directory of this project.
  */
-import {AfterViewInit, Directive, ElementRef, HostListener} from "@angular/core";
+import {AfterViewInit, Directive, ElementRef, HostListener, Input, OnDestroy} from "@angular/core";
+import {FocusTrapLinker} from "./focus-trap-linker.service";
 
 @Directive({selector: "[clrFocusTrap]"})
-export class FocusTrapDirective implements AfterViewInit {
-    constructor(public elementRef: ElementRef) {}
+export class FocusTrapDirective implements AfterViewInit, OnDestroy {
+    prevFocusTrap: FocusTrapDirective;
+    focusTrapEnabled: boolean;
+
+    constructor(public elementRef: ElementRef, private focusTrapLinker: FocusTrapLinker) {
+        if (this.focusTrapLinker.currentFocusTrap && this.focusTrapLinker.currentFocusTrap.focusTrapEnabled) {
+            this.focusTrapLinker.currentFocusTrap.focusTrapEnabled = false;
```

Comment:
I like the idea, but the execution here is messy. I mean, let's just phrase the current structure:
"Directives are directly setting public properties of other directives that they grab through a service to which said directives registered manually."
This is convoluted and will lead to bug, if you don't already have some. 

What you want is to move all of that logic to the service, and only do a basic registration here.
For instance, you can expose a method like this in the service:
```typescript
export class FocusTrapLinker {
    ...
    isEnabled(focusTrap: FocusTrapDirective): boolean {
        // Change this once you move the service to an array of focus traps, as mentioned in my other comment
        return this. currentFocusTrap === focusTrap;
    }
}
```

Reply:
I know this is super messy currently. I just wanted to get a feeling of your perspectives on this approach first. I will improve the overall code quality in this PR and include the tests in it.